### PR TITLE
Fix warnings (causing CI failures) with new Rust beta.

### PR DESCRIPTION
**Pull request by cfallin**
- Panic messages must now be string literals (we used `format!()` in
  many places; `panic!()` can take format strings directly).
- Some dead enum options with EVEX encoding stuff in old x86 backend.
  This will go away soon and/or be moved to the new backend anyway, so
  let's silence the warning for now.
- A few other misc warnings.

We're seeing these warnings in CI for #2653.

### CODE REVIEW

**Comment by alexcrichton** [comment]
I think a number of the panic/format-related message warnings were fixed in https://github.com/bytecodealliance/wasmtime/pull/2652, but the other changes look good here! Want to update the `rand` crate as well to fix the `cargo deny` error?

**Comment by cfallin** [comment]
Ah yes, updated for `cargo-deny` as well, thanks!

**Review by alexcrichton — APPROVED** [review]

**Comment by cfallin** [comment]
Hmm, now `cargo-audit` is unhappy but it's unclear where the old `rand_core` is coming from -- investigating...

**Comment by cfallin** [comment]
OK, so removing `Cargo.lock` and attempting a rebuild runs into some errors in `bitvec` conflicting with `funty` (both have traits that define a `BITS` associated constant and there's a resolution conflict). This is getting... exciting. I'm not sure how to proceed generating a fresh `Cargo.lock` without opening some PRs upstream.

If I try to just trim our `Cargo.lock` in a more scoped way, I see that the bad version `rand_core 0.5.1` is used by `wasi-crypto`. @sunfishcode or @pchickey, thoughts on this?

(Separately, I have some concerns about random CI breakages that require upstream fixes and cause our work to halt -- do we want to talk about whether `cargo-audit` / `cargo-deny` should be warnings-we-address-ASAP rather than red-CI-errors? Are there compliance reasons they have to be the latter? @tschneidereit thoughts?)

File: cranelift/codegen/src/isa/x86/binemit.rs [hunk 1]
```diff
@@ -296,6 +297,7 @@ impl EvexVectorLength {
 }
 
 /// The EVEX format allows defining rounding control in the `L'` and `L` bits; see `EvexContext`.
+#[allow(dead_code)]
```
Comment (abrown) [inline cranelift/codegen/src/isa/x86/binemit.rs:300]: 👍

**Comment by cfallin** [comment]
Closing as #2656 incorporates these fixes along with fixes for the cargo-audit reports.

**Pull request closed without merging**